example/src/main/java/io/netty/example/http2/tiles/http2/ImageCache.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.example.http2.tiles.http2;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+
+/**
+ * Caches the images to avoid reading them every time from the disk.
+ */
+public final class ImageCache {
+
+    private static ImageCache INSTANCE;
+
+    private final Map<String, byte[]> imageBank = new HashMap<String, byte[]>(200);
+
+    private ImageCache() {
+    }
+
+    public static synchronized ImageCache instance() {
+        if (INSTANCE == null) {
+            INSTANCE = new ImageCache();
+            INSTANCE.initImageBank();
```

Comment:
Probably, yes. I just to try to avoid having a dependency on external entities (file system here) while constructing the object. But here, indeed, we're in a synchronized block and the object won't escape until it's properly constructed anyway. So, yes, I could move the code inside the constructor.